enhancements/cluster-logging/cluster-logging-kibana-multitenancy.md

```diff
@@ -0,0 +1,205 @@
+an user are defined within the user roles that are seeded into Elasticsearch. This functionality is to be provided by the Elasticsearch Proxy.
+
+### User Stories
+
+#### As an OKD admin, I want to be able to view all logs that I have access to
+
+This is the current state, admin users should still be able to view all the cluster logs with the provided index patterns
+
+#### As an OKD user, I want to be able to view all container logs form the namespaces to which I have access
+
+This is current state, however instead of an index pattern per namespace the user has access to, it will be a single static index pattern with access controlled by DLS.
+
+#### As an OKD user, I want to be able to create and share visualizations for logs
+
+Currently users are unable to share their visualizations since Kibana did not have knowledge on how to access the specific user's Kibana index where the visualizations were stored. They would be able to share them now as Kibana specifies the tenant as part of the visualization path when sharing.
+
+### Implementation Details
+
+#### Assumptions
+
+* The Elasticsearch Proxy will configure roles for users such that they have a tenant pattern defined that matches what Kibana would create for an user logging in.
+* Each user would have only one tenant available to them: their user name based tenant
+
+#### Security
+
+No additional security concerns need to be addressed by this plugin.
+
+### Risks and Mitigations
+
+## Design Details
+
+As part of the Kibana6 image building and distribution, we will provide the OpenDistro plugin for Kibana.
+Installing and configuring that plugin will allow us to provide multitenancy from within Kibana.
+We want to avoid using the `private` tenant as it would be a duplicate of the user tenant and may be confusing when trying to share a visualization.
+
+### kibana.conf
+
+As part of configuring multitenancy for Kibana, we would need to add the following to the Kibana config:
+
+```
+elasticsearch.requestHeadersWhitelist: ["securitytenant","Authorization"]
+opendistro_security.multitenancy.enabled: true
+opendistro_security.multitenancy.tenants.enable_global: false
+opendistro_security.multitenancy.tenants.enable_private: false
+opendistro_security.multitenancy.tenants.preferred: ["admin"]
+opendistro_security.multitenancy.enable_filter: false
+```
+
+### roles.yml
+
+In addition to needing to configure Kibana to allow multitenancy, we need to update the Elasticsearch `roles.yml` to specify `tenant_permissions` for each user's role.
+
+```
+example-role:
+  reserved: false
+  hidden: false
+  cluster_permissions:
+  - "read"
+  - "cluster:monitor/nodes/stats"
+  - "cluster:monitor/task/get"
+  index_permissions:
+  - index_patterns:
+    - "apps.*"
```

Comment:
Since we are using aliases ad defined https://github.com/openshift/enhancements/blob/master/enhancements/cluster-logging/cluster-logging-es-rollover-data-design.md#data-model I don't think we need wild cards.